File: java/src/RegexUtilities.java
import java.util.regex.Pattern;

public class RegexUtilities {
    public static boolean isValidEmail(String email) {
        if (email == null || email.isEmpty() || email.isBlank()) {
            return false;
        }

        // A valid email address pattern from OWASP Validation Regex repository
        String emailRegex = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,}$";
        Pattern pattern = Pattern.compile(emailRegex);
        return pattern.matcher(email).matches();
    }

    public static boolean isValidName(String name) {
        if (name == null || name.isEmpty() || name.isBlank())
            return false;

        String nameRegex = "^[a-zA-Z \\-\\.\\']*$";
        Pattern pattern = Pattern.compile(nameRegex);
        return pattern.matcher(name).matches();
    }

    public static Boolean isValidDouble(String doubleString)
        {
            double balance = Double.MIN_VALUE;
            try
            {
                balance = Double.parseDouble(doubleString);
            }
            catch (NumberFormatException exception)
            {
                System.out.println("The input does not represent a number in valid format.");
                return false;
            }
            catch (NullPointerException exception)
            {
                System.out.println("The input is null or empty.");
                return false;
            }
            if (balance < 0)
            {
                System.out.println("The input cannot be negative.");
                return false;
            }
            return true;
        }
}

File: core/RegularExpressions/RegexUtilities.cs
using System;
using System.Globalization;
using System.Text.RegularExpressions;

namespace Banking
{
    class RegexUtilities
    {
        public static Boolean IsValidEmail(String email)
        {
            if (String.IsNullOrWhiteSpace(email))
                return false;

            try
            {
                // A valid email address pattern from Microsoft Learn (.NET)
                return Regex.IsMatch(email, @"^[^@\s]+@[^@\s]+\.[^@\s]+$", RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(250));
            }
            catch (RegexMatchTimeoutException)
            {
                return false;
            }
        }
        public static Boolean IsValidName(String name)
        {
            if (String.IsNullOrEmpty(name))
                return false;

            return Regex.IsMatch(name, @"^[\p{L}\p{M}' \.\-]+$");
        }
        public static Boolean IsValidDouble(String doubleString)
        {
            Double balance = Double.MinValue;
            try
            {
                balance = Double.Parse(doubleString);
            }
            catch (OverflowException)
            {
                Console.WriteLine($"The input is outside the range of the Double type!");
                return false;
            }
            catch (FormatException)
            {
                Console.WriteLine($"The input does not represent a number in valid format.");
                return false;
            }
            catch (ArgumentNullException)
            {
                Console.WriteLine($"The input is null or empty.");
                return false;
            }
            if (balance < 0)
            {
                Console.WriteLine($"The input cannot be negative.");
                return false;
            }
            return true;
        }
    }
}